packages/devtools_app/lib/src/memory/flutter/memory_filter.dart

```diff
@@ -0,0 +1,391 @@
+
+    filteredGroup.removeWhere((groupName, libraryFilter) {
+      if (dartGlobal &&
+          groupName != _prettyPrintDartAbbreviation &&
+          groupName.startsWith(_dartLibraryUriPrefix)) {
+        return true;
+      } else if (flutterGlobal &&
+          groupName != _prettyPrintFlutterAbbreviation &&
+          groupName.startsWith(_flutterLibraryUriPrefix)) {
+        return true;
+      }
+
+      return false;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    buildFilters();
+
+    return Dialog(
+      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(20.0)),
+      child: LayoutBuilder(
+        builder: (BuildContext context, BoxConstraints constraints) {
+          return Container(
+            width: MediaQuery.of(context).size.width / 3,
+            height: constraints.maxHeight < 500
+                ? constraints.maxHeight
+                : constraints.maxHeight / 3 + 300,
+            child: Padding(
+              padding: const EdgeInsets.only(left: 15),
+              child: Column(
+                mainAxisAlignment: MainAxisAlignment.center,
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  Column(
+                    children: [
+                      const TextField(
+                        decoration: InputDecoration(
+                            border: InputBorder.none,
+                            isDense: true,
+                            labelText: 'Filter Snapshot'),
+                      ),
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.start,
+                        children: [
+                          Checkbox(
+                              value: controller.filterPrivateClasses,
+                              onChanged: (value) {
+                                setState(() {
+                                  controller.filterPrivateClasses = value;
+                                });
+                              }),
+                          const Text('Hide Private Class e.g.,_className'),
+                        ],
+                      ),
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.start,
+                        children: [
+                          Checkbox(
+                              value: controller.filterZeroInstances,
+                              onChanged: (value) {
+                                setState(() {
+                                  controller.filterZeroInstances = value;
+                                });
+                              }),
+                          const Text('Hide Classes with No Instances'),
+                        ],
+                      ),
+                      Row(
+                        mainAxisAlignment: MainAxisAlignment.start,
+                        children: [
+                          Checkbox(
```

Comment:
Can you add a ValueListenableCheckbox class to our shared UI library?
That would reduce the boilerplate for this code and would be handy elsewhere.

Reply:
Yep, it was a bit tricky (since Checkbox is immutable) - but level of indirection solved the problem.

```
/// Checkbox Widget class using a ValueNotifier.
class NotifierCheckbox extends Checkbox {
  NotifierCheckbox({
    Key key,
    @required this.notifier,
  }) : super(
          key: key,
          value: notifier.value,
          onChanged: notifier.onChanged,
        );

  final CheckboxValueNotifier notifier;

  /// Notifies that the value of the checkbox has changed.
  ValueListenable<bool> get valueListenable => notifier;
}

/// ValueNotifier associated with a Checkbox widget.
class CheckboxValueNotifier extends ValueNotifier<bool> {
  CheckboxValueNotifier(bool value) : super(value);

  void onChanged(bool newValue) {
    value = newValue;
  }
}

```